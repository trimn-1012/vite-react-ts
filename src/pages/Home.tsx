import HomeTemplate from '../templates/Home';

const Home = () => {
  return (
    <div>
      Home Page
      <HomeTemplate />
    </div>
  );
};

export default Home;
