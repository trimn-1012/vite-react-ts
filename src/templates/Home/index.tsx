import { FC, PropsWithChildren } from 'react';

const HomeTemplate: FC<PropsWithChildren> = () => {
  return <div>Home Template</div>;
};

export default HomeTemplate;
