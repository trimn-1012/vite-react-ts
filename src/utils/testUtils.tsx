import { render, RenderOptions } from '@testing-library/react';
import { FC, ReactElement, PropsWithChildren } from 'react';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import { SWRConfig } from 'swr';

import ReduxProvider from '@/providers/Redux';

export function renderWithReduxProvider(
  ui: ReactElement,
  renderOptions: RenderOptions = {},
) {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => (
    <ReduxProvider>{children}</ReduxProvider>
  );

  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export function renderWithSWRConfig(
  ui: ReactElement,
  renderOptions: RenderOptions = {},
) {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => (
    <SWRConfig
      value={{
        errorRetryInterval: 3000,
        errorRetryCount: 3,
        provider: () => new Map(),
      }}
    >
      {children}
    </SWRConfig>
  );

  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export function renderWithRouter(
  ui: React.ReactElement,
  {
    initialEntries,
    ...renderOptions
  }: {
    initialEntries?: MemoryRouterProps['initialEntries'];
  } & RenderOptions = {},
) {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => (
    <MemoryRouter initialEntries={initialEntries}>{children}</MemoryRouter>
  );

  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export function renderWithAllProviders(
  ui: React.ReactElement,
  {
    initialEntries,
    ...renderOptions
  }: {
    initialEntries?: MemoryRouterProps['initialEntries'];
  } & RenderOptions = {},
) {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => (
    <ReduxProvider>
      <SWRConfig
        value={{
          errorRetryInterval: 3000,
          errorRetryCount: 3,
          provider: () => new Map(),
        }}
      >
        <MemoryRouter initialEntries={initialEntries}>{children}</MemoryRouter>
      </SWRConfig>
    </ReduxProvider>
  );

  return render(ui, { wrapper: Wrapper, ...renderOptions });
}
