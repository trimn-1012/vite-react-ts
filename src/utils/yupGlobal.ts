import * as yup from 'yup';

// add more method validate here
// yup.addMethod<yup.StringSchema>(
//   yup.string,
//   'password',
//   function (message: string) {
//     return this.matches(Regex.PASSWORD, {
//       message,
//       excludeEmptyString: true,
//     });
//   },
// );

export default yup;
