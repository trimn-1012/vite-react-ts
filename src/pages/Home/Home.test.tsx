import { renderWithAllProviders } from '@/utils/testUtils';

import Home from '.';

describe('pages/Home', () => {
  describe('snapshot', () => {
    it('should render and match the snapshot', () => {
      const { container } = renderWithAllProviders(<Home />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
