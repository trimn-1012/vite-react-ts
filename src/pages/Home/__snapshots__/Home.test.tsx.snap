// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pages/Home > snapshot > should render and match the snapshot 1`] = `
<div>
  Home Page
  <div>
    Home Template
  </div>
</div>
`;
